nav2_tasks/include/nav2_tasks/navigate_to_pose_action.hpp

```diff
@@ -0,0 +1,64 @@
+// Copyright (c) 2018 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef NAV2_TASKS__NAVIGATE_TO_POSE_ACTION_HPP_
+#define NAV2_TASKS__NAVIGATE_TO_POSE_ACTION_HPP_
+
+#include <string>
+#include <memory>
+#include "nav2_tasks/bt_conversions.hpp"
+#include "nav2_tasks/bt_action_node.hpp"
+#include "nav2_tasks/navigate_to_pose_task.hpp"
+#include "geometry_msgs/msg/point.hpp"
+#include "geometry_msgs/msg/quaternion.hpp"
+
+namespace nav2_tasks
+{
+
+class NavigateToPoseAction : public BtActionNode<NavigateToPoseCommand, NavigateToPoseResult>
+{
+public:
+  NavigateToPoseAction(const std::string & action_name, const BT::NodeParameters & params)
+  : BtActionNode<NavigateToPoseCommand, NavigateToPoseResult>(action_name, params)
+  {
+    // Create the input and output messages
+    command_ = std::make_shared<nav2_tasks::NavigateToPoseCommand>();
+    result_ = std::make_shared<nav2_tasks::NavigateToPoseResult>();
+
+    // Use the position and orientation fields from the behavior tree node parameter
```

Comment:
This section of code looks the same as the ComputePathToPose action. Can we factor it out?

Reply:
They're a bit different (now that I've finished implementing the passing of attributes). The NavigateToPose takes the goal pose, while the ComputePathToPose takes the starting pose and the goal pose. Not sure about sharing at this point.